parser.go

```diff
@@ -914,6 +916,19 @@ func ArrayEach(data []byte, cb func(value []byte, dataType ValueType, offset int
 	offset = 1
 
 	if len(keys) > 0 {
+		// Detects nil in the path given
+		// for i, v := range data {
+		//fmt.Printf("index: [%d]   value: %s   address: [%v]\n", i, string(v), &v)
+		// if &i == nil {
+		// 	fmt.Println("YAY")
+		// 	return offset, ArrayEachNullError
+		// }
+		//}
+
+		if strings.Contains(string(data), "nil") {
```

Comment:
Would we want to loop over each item in the array so that we can provide a verbose error? I.e. key: [derp] is null